plugins/tiddlywiki/filesystem/filesystemadaptor.js

```diff
@@ -60,7 +66,23 @@ FileSystemAdaptor.prototype.getTiddlerFileInfo = function(tiddler,callback) {
 		});
 		$tw.boot.files[title] = fileInfo;
 	}
-	callback(null,fileInfo);
+	else if(fileInfo && fileSystemPaths) {
+		// Otherwise (check for `FileSystemPaths`), we'll need to (re)generate it
+		options.fileInfo = {...fileInfo};
+		fileInfo = $tw.utils.generateTiddlerFileInfo(tiddler,{
+			directory: $tw.boot.wikiTiddlersPath,
+			pathFilters: this.wiki.getTiddlerText("$:/config/FileSystemPaths","").split("\n"),
+			wiki: this.wiki,
+			fileSystemPath: options.fileInfo.filepath
+		});
+		if(options.fileInfo && options.fileInfo.filepath == fileInfo.filepath) {
+			options = null;
```

Comment:
should it be `null` or `{}`

Reply:
Thanks very much for the additional eyes on this!

options was added to store a copy of the fileInfo to test against the regenerated fileInfo. If the `filepath` property matches, the file is going to be written to the same disk location, so options is set to null.

If the path has changed, the new fileInfo is stored and the existence of options triggers the additional `self.deleteTiddler` callback (your comment below) using the options object to tell it where to delete the old file-copy from.